### PR TITLE
Adding tests requested in #295 

### PR DESCRIPTION
**Pull request by petedmarsh**
I recent raised #295 to fix a bug, this is a bug that could have been caught with tests but no tests had been written (I wrote the original code, so it is my fault).

I've had a stab here at writing some tests for this case, they required a small amount of refactoring. I fully expect that I can improve upon them and I am happy to do so. For one thing the test file is in the main `src` tree rather than `test` as it seemed silly to me to make these functions public, but it also seems silly to have a test here.

Anyway, I am happy to improve and add more, I am raising early in the hope I can get some feedback (though I appreciate that that isn't your job to do so!).

### CODE REVIEW

**Comment by petedmarsh** [comment]
This'll need to be rebased anyway as 361e3ed is in main under a different SHA, I will sort that out.

**Comment by mattklein123** [comment]
cc @nezdolik @ysawa0 for comments.

**Comment by nezdolik** [comment]
Thanks, I will review this week.

File: src/srv/srv.go [hunk 1]
```diff
@@ -21,7 +27,7 @@ func ParseSrv(srv string) (string, string, string, error) {
 	return matches[1], matches[2], matches[3], nil
 }
 
-func ServerStringsFromSrv(srv string) ([]string, error) {
+func (s DnsSrvResolver) ServerStringsFromSrv(srv string) ([]string, error) {
```
Comment (nezdolik) [inline src/srv/srv.go:30]: nit: use more descriptive var name instead of `s`

File: src/memcached/cache_impl_test.go [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+	serverList.Each(func(addr net.Addr) error {
+		actualMemcacheHosts = append(actualMemcacheHosts, addr.String())
+		return nil
+	})
+
+	assert.Equal([]string{mockMemcacheHostPort}, actualMemcacheHosts)
+}
+
+func TestRefreshServerSetsServersOnNonEmptyServerList(t *testing.T) {
```
Comment (nezdolik) [inline src/memcached/cache_impl_test.go:41]: nit: i think more descriptive name this test case would be `...OverridesServersOnNonEmptyServerList`

**Comment by nezdolik** [comment]
Thank you for adding the tests! Few nits from my side, and looks like commits are missing sign-off.

**Comment by petedmarsh** [comment]
@nezdolik I have updated everything as per your comments

File: src/srv/srv.go (hunk 1 above)
Comment (nezdolik) [reply src/srv/srv.go:30]: @petedmarsh ping on this

**Comment by nezdolik** [comment]
@petedmarsh looks like one more nit left and then you can mark this PR as ready for review (instead of Draft)

**Review by nezdolik — APPROVED** [review]

**Review by mattklein123 — APPROVED** [review]
Awesome, thanks!

**Pull request merged**